pkg/vsphere/disk/disk.go

```diff
@@ -45,6 +45,43 @@ type Filesystem interface {
 	Unmount(path string) error
 }
 
+// ReferenceCount represents the number of references to a disk
+type ReferenceCount struct {
```

Comment:
This is an unbounded counting semaphore - can see why we're not implementing it using channels, but calling it a semaphore might be less confusing.